_posts/2016-08-18-On-Emission.markdown

```diff
@@ -0,0 +1,130 @@
+---
+layout: post
+title: "On our implementation of React Native"
+date: 2016-08-18 12:17
+author: orta
+categories: [tooling, mobile, eigen, node, reactnative]
+series: React Native at Artsy
+---
+
+<center>
+ <img src="/images/emission/emission-logo-artsy.svg" style="height:300px;">
+</center>
+
+I arrived fashionably late to the React Native party in Artsy. I had been a part of our [Auctions Team][auctions_team], where we worked in Swift with [some light-FRP][interstellar]. We were not affected by 4 months of work on moving to React Native, at all. 
+
+It was a quiet revolution. We did not have to install `npm`, made zero changes to the code for auctions and the whole app's infrastructure barely changed. What gives? 
+
+Well, first up we weren't planning a re-write, we don't have that kind of luxury and the scope of our app is too big compared to the team working on it. Second, we reused existing dependency infrastructure. Read on to find out what that looks like.
+
+<!-- more -->
+
+### Why we were in a good position to do this
+
+Let's talk a little about the Artsy flagship app, [Eigen][eigen]. It's an app that aimed to comprehensively cover the art world. From [Shows](https://www.artsy.net/shows) to [Galleries](https://www.artsy.net/galleries), [Fairs](https://www.artsy.net/art-fairs), [Auctions](https://www.artsy.net/auctions), [Museums & Institutions](https://www.artsy.net/institutions).  
+
+It all looks a bit like this: 
+
+{% expanded_img /images/emission/eigen-overview.jpg %}
+
+Our app neatly splits into two areas of view controllers, ones that act as a browser chrome, and individual view controllers that normally map 1:1 to routes on the Artsy website. 
+
+For example, `artsy.net/artwork/glenn-brown-suffer-well` maps to the native `ARArtworkViewController`. 
+
+{% expanded_img /images/emission/eigen.svg %}
+
+Just as a browser knows very little about the individual content of the pages that it is rendering, the eigen chrome exists _relatively_ independent of the view controllers that are showing. 
+
+Each view controller also knows very little about each other, so actions that trigger a new view controller are generally done by creating a string route and passing it through the routing system. I've wrote about about this in [Cocoa Architecture: Router Pattern][router_pattern]. 
+
+Interestingly if the router cannot route a view controller, it will pass through to a web view. This is why we consider the app a [hybrid app][hybrid_app]. This system means adding new view controllers is extremely easy.
+
+### Introducing Emission
+
+Emission is:
+
+- A node module.
+- A CocoaPod.
+- An iOS App.
+
```

Comment:
Maybe add a paragraph explaining what it is in context to Eigen, e.g.:

"Emission is what we use to contain all of our React Native components in, that our main app Eigen can depend on and use, without needing to bother with the internals / details of React Native, Javascript, and so on."